### PR TITLE
feat(katana): db storage intergration

### PR DESCRIPTION
**Pull request by kariy**
Depends on #1436 

This PR adds persistent db as one of the available options for `katana` storage implementations, along with in-memory and forked db. 

- Adds `--db-dir <PATH>` CLI arg. for specifying where the DB directory will be located.

### CODE REVIEW

File: crates/katana/core/src/backend/storage.rs [hunk 1]
```diff
@@ -207,4 +229,106 @@ mod tests {
         assert_eq!(header.parent_hash, FieldElement::ZERO);
         assert_eq!(block_status, FinalityStatus::AcceptedOnL1);
     }
+
+    #[test]
+    fn blockchain_from_dn() {
```
Comment (glihm) [inline crates/katana/core/src/backend/storage.rs:234]:
```suggestion
    fn blockchain_from_db() {
```

File: crates/katana/storage/db/src/utils.rs [hunk 2]
```diff
@@ -16,8 +16,10 @@ pub(crate) fn default_page_size() -> usize {
     os_page_size.clamp(MIN_PAGE_SIZE, LIBMDBX_MAX_PAGE_SIZE)
 }
 
-/// Check if a db is empty. It does not provide any information on the
-/// validity of the data in it. We consider a database as non empty when it's a non empty directory.
+/// Check if a Db is empty. We consider a database as non empty when `path` it's a non empty
+/// directory or the `path` exists.
```
Comment (glihm) [inline crates/katana/storage/db/src/utils.rs:20]: Not sure if this matches the actual behavior. The code returns true (empty) if the path does not exist.

**Review by glihm — APPROVED** [review]
Minor comment on typo and comment.
This PR is the achievement of a great work on persistence @kariy, well done! 🔥

File: crates/katana/storage/db/src/utils.rs (hunk 2 above)
Comment (kariy) [reply crates/katana/storage/db/src/utils.rs:20]: ah yes it should be ".. or if the `path` does not exist. "

**Pull request merged**